### PR TITLE
[Ad] Implement targeting the ads. Passing `params`

### PR DESCRIPTION
**Pull request by asimqt**
@rvgpl @ajaymenon0 Please review.

### CODE REVIEW

File: app/isomorphic/components/dfp-ads.js [hunk 1]
```diff
@@ -9,11 +11,31 @@ export const AD_CONFIG = {
...
-  defaultNetworkID: "60988533",
-  config: AD_CONFIG,
-  targeting: function(state) {
-    const params = {};
-    return params;
-  }
-});
+export function DfpAd(props) {
+  const DfpAdWrapper = createDfpAdComponent({
+    defaultNetworkID: "60988533",
```
Comment (rvgpl) [inline app/isomorphic/components/dfp-ads.js]: Why hard code this? Pick it from config.

File: app/isomorphic/components/dfp-ads.js [hunk 2]
```diff
@@ -9,11 +11,31 @@ export const AD_CONFIG = {
...
+    config: AD_CONFIG,
+    targeting: function(state) {
+      // Params that can be derived from state.
+      const params = {
+        pageType: _.get(state, ['qt', 'pageType']),
+        publisherName: _.get(state, ['qt', 'config', 'publisher-name']),
```
Comment (rvgpl) [inline app/isomorphic/components/dfp-ads.js]: Pass publisher id also. Easy to target

File: app/isomorphic/components/dfp-ads.js [hunk 3]
```diff
@@ -9,11 +11,31 @@ export const AD_CONFIG = {
...
+      const storyId = _.get(state, ['qt', 'data', 'story', 'id']);
+      if(storyId) params['storyId'] = storyId;
+      const sponsoredBy = _.get(state, ['qt', 'data', 'story', 'metadata','sponsored-by']);
+      if(sponsoredBy) params['sponsoredBy'] = sponsoredBy;
+
+      // Params that can't be derived from state, thus using props to pass it on.
+      const layoutName = _.get(props, 'layoutName');
+      if(layoutName) params['layoutName'] = layoutName;
+      const collectionName = _.get(props, 'collectionName');
+      if(collectionName) params['collectionName'] = collectionName;
```
Comment (rvgpl) [inline app/isomorphic/components/dfp-ads.js]: Add collection id also for more finer control over targetting

File: app/isomorphic/components/layout/l-shape-one-widget.js [hunk 4]
```diff
@@ -34,7 +34,7 @@ function LShapeOneWidget(props) {
           <StoryCard story={props.stories[0]} config={primarStoryCardConfig} type="imageBackground"/>
         </div>
         <div className="app-ad app-ad--vertical">
-          <DfpAd adtype="Vertical-Ad"/>
+          <DfpAd adtype="Vertical-Ad" layoutName="LShapeOneWidget" collectionName={props.config['collection-name']}/>
```
Comment (rvgpl) [inline app/isomorphic/components/layout/l-shape-one-widget.js]: Don't you think it is better to pass collection itself and pick id and name in the DfpAd component?

File: app/isomorphic/components/dfp-ads.js (hunk 1 above)
Comment (asimqt) [reply app/isomorphic/components/dfp-ads.js]: Will do it in a new PR. This PR targets to pass targeting params only

File: app/isomorphic/components/dfp-ads.js (hunk 2 above)
Comment (asimqt) [reply app/isomorphic/components/dfp-ads.js]: ok

File: app/isomorphic/components/dfp-ads.js (hunk 3 above)
Comment (asimqt) [reply app/isomorphic/components/dfp-ads.js]: Do you think that much fine tuning is necessary? For ex:, in home page, we are passing `layoutName` and `collectionName` which I think is necessary. Adding more params will only increase complexity. Can you think of a scenario where we need `id`?

File: app/isomorphic/components/layout/l-shape-one-widget.js (hunk 4 above)
Comment (asimqt) [reply app/isomorphic/components/layout/l-shape-one-widget.js]: Ye. You are right. But I'm not sure about passing `id`.

**Pull request merged**